digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/masstransitstop/ServicePointStopService.scala

```diff
@@ -0,0 +1,127 @@
+
+  def withId(id: Long)(query: String): String = {
+    query + s" and a.id = $id"
+  }
+
+  def withNationalId(id: Long)(query: String): String = {
+    query + s" and a.external_id = $id"
+  }
+
+  def create(lon: Double, lat: Double, properties: Seq[SimplePointAssetProperty], username: String, municipalityCode: Int, withTransaction: Boolean = true): Long = {
+    def createAssetAndProperties = {
+      val assetId = Sequences.nextPrimaryKeySeqValue
+      servicePointBusStopDao.insertAsset(assetId, lon, lat, username, municipalityCode)
+      servicePointBusStopDao.updateAssetProperties(assetId, properties)
+      assetId
+    }
+
+    if (withTransaction) {
+      withDynTransaction {
+        createAssetAndProperties
+      }
+    } else {
+      createAssetAndProperties
+    }
+  }
+
+  def getById(id: Long): Option[ServicePoint] = {
+    withDynSession {
+      servicePointBusStopDao.fetchAsset(withId(id)).headOption
+    }
+  }
+
+  def getByNationalId(nationalId: Long): Option[ServicePoint] = {
+    withDynSession {
+      servicePointBusStopDao.fetchAsset(withNationalId(nationalId)).headOption
+    }
+  }
+
+  def fetchAsset(id: Long): ServicePoint = {
+    servicePointBusStopDao.fetchAsset(withId(id)).headOption.getOrElse(throw new NoSuchElementException)
+  }
+
+  def expire(asset: ServicePoint, username: String, withTransaction: Boolean = true) = {
+    if (withTransaction) {
+      withDynTransaction {
+        servicePointBusStopDao.expire(asset.id, username)
+      }
+    } else {
+      servicePointBusStopDao.expire(asset.id, username)
+    }
+  }
+
+  protected def updatePosition(id: Long, position: PositionCoordinates, municipalityCode: Int) = {
+    val point = Point(position.lon, position.lat)
+    servicePointBusStopDao.updateMunicipality(id, municipalityCode)
+    updateAssetGeometry(id, point)
+  }
+
+
+  def enrichBusStop(asset: ServicePoint): (ServicePoint, Boolean) = {
+    (asset.copy(propertyData = asset.propertyData), false)
+  }
+
+  def withFilter(filter: String)(query: String): String = {
+    query + " " + filter
+  }
+
+  def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[ServicePoint] = {
+    withDynSession {
+      val boundingBoxFilter = OracleDatabase.boundingBoxFilter(bounds, "a.geometry")
+      val filter = s"and a.asset_type_id = ${MassTransitStopAsset.typeId} and $boundingBoxFilter"
+      servicePointBusStopDao.fetchAsset(withFilter(filter))
+    }
+  }
+
+  def getByMunicipality(municipalityCode: Int): Seq[ServicePoint] = {
+    withDynSession {
+      val filter = s"and a.asset_type_id = ${MassTransitStopAsset.typeId}  and a.municipality_code = $municipalityCode"
+      servicePointBusStopDao.fetchAsset(withFilter(filter))
+    }
+  }
+
+  def update(assetId: Long, position: PositionCoordinates, properties: Seq[SimplePointAssetProperty], username: String, municipalityCode: Int, withTransaction: Boolean = true): ServicePoint = {
+    def update = {
```

Comment:
Let's switch the name of this to something different from the main method.

Reply:
Something like auxUpdate?

Reply:
@marquesfm Applied last suggestion